migration/orm_test.go

```diff
@@ -0,0 +1,154 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestSchemaVersionedBucket(t *testing.T) {
+	const thisPkgName = "testpkg"
+
+	reg := newRegister()
+
+	reg.MustRegister(1, &MyModel{}, NoModification)
+	reg.MustRegister(2, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		msg := m.(*MyModel)
+		msg.Cnt += 2
+		return msg.err
+	})
+	reg.MustRegister(3, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		panic("not implemented")
+	})
+
+	db := store.MemStore()
+
+	schema := NewSchemaBucket()
+	if _, err := schema.Create(db, &Schema{Metadata: &weave.Metadata{Schema: 1}, Pkg: thisPkgName, Version: 1}); err != nil {
+		t.Fatalf("cannot register schema version: %s", err)
+	}
+
+	b := &MyModelBucket{
+		Bucket: NewBucket(thisPkgName, "mymodel", orm.NewSimpleObj(nil, &MyModel{})),
```

Comment:
And this uses `migration.Bucket` instead of `orm.Bucket`, right?
With just one more argument so it is package-schema aware